### PR TITLE
Add build-provenance in publish-commit-bottles.yml

### PR DESCRIPTION
**Pull request by josephsweeney**
Adds an extra step after pr-pull in publish-commit-bottles.yml to generate build provenance for bottles that get published.

This requires a small change in `Homebrew/brew` to retain the temporary directory and to expose the path to the following workflow steps. [This PR](https://github.com/Homebrew/brew/pull/16530) must be merged first before this change will work.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by josephsweeney** [comment]
I will wait to mark this as ready for review until the [PR mentioned above](https://github.com/Homebrew/brew/pull/16530) is merged.

In the meantime, if @Bo98, @carlocab, @MikeMcQuaid, @woodruffw have any feedback about testing this or the above PR, please let me know!

**Review by MikeMcQuaid — APPROVED** [review]
Makes sense to me! Happy for this to be merged as-is and reverted if it causes issues.

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by josephsweeney** [comment]
> Makes sense to me! Happy for this to be merged as-is and reverted if it causes issues.

That works for me, we'll definitely have to keep an eye out after merging on the [publish-commit-bottles action](https://github.com/Homebrew/homebrew-core/actions/workflows/publish-commit-bottles.yml) but if I missed anything, it should be obvious.

File: .github/workflows/publish-commit-bottles.yml [hunk 1]
```diff
@@ -316,12 +318,18 @@ jobs:
...
             ${{inputs.warn_on_upload_failure && '--warn-on-upload-failure' || ''}} \
             ${{inputs.message && '--message="$INPUT_MESSAGE"' || ''}} \
             "$PR"
 
           echo "head_sha=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
 
+      - name: generate build provenance
+        uses: github-early-access/generate-build-provenance@main
+        with:
+          subject-path: '${{steps.pr-pull.outputs.bottle-path}}/*.tar.gz'
```
Comment (Bo98) [inline .github/workflows/publish-commit-bottles.yml:331]:
```suggestion
          subject-path: '${{steps.pr-pull.outputs.bottle-path}}/*.tar.gz'
```

This says `bottle-path` but pr-pull.rb has `bottle_path`.

Comment (josephsweeney) [reply .github/workflows/publish-commit-bottles.yml]: Thank you for catching this! I wish I had some way to run this to catch mistakes like this. But you saved us a cycle of committing and reverting, so thank you 😅

Comment (MikeMcQuaid) [reply .github/workflows/publish-commit-bottles.yml]: `actionlint` may catch some of this stuff. we should consider using it on this repo IMO.

Comment (Bo98) [reply .github/workflows/publish-commit-bottles.yml]: `actionlint` probably doesn't know much about what the Ruby side does

**Review by Bo98 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
Let's try this...

**Comment by carlocab** [comment]
Ok, I guess that worked? https://github.com/Homebrew/homebrew-core/actions/runs/7679301023/job/20930044698#step:9:1

**Comment by josephsweeney** [comment]
Thank you for merging @carlocab, it looks like it is working! I can see the attestation page you linked. I think that because the feature is still in early access, they haven't rolled it out to everyone yet.

**Comment by woodruffw** [comment]
Great work here @josephsweeney! 

@carlocab: I'll see about getting you added to the private beta (along with the other org members) 🙂

**Comment by woodruffw** [comment]
@carlocab You should have access to those URLs now!

(I've also asked them to add @MikeMcQuaid and @Bo98 initially -- adding everyone from the Homebrew org is manual on their side, but anybody else who wants early access can be added!)

**Comment by p-linnane** [comment]
> @carlocab You should have access to those URLs now!
> 
> (I've also asked them to add @MikeMcQuaid and @Bo98 initially -- adding everyone from the Homebrew org is manual on their side, but anybody else who wants early access can be added!)

Put me in coach!

**Comment by MikeMcQuaid** [comment]
> (I've also asked them to add @MikeMcQuaid and @Bo98 initially -- adding everyone from the Homebrew org is manual on their side, but anybody else who wants early access can be added!)

@josephsweeney Me too, please!

**Comment by josephsweeney** [comment]
@MikeMcQuaid I believe you have been added. Can you see the attestations [here](https://github.com/Homebrew/homebrew-core/attestations)?

**Comment by MikeMcQuaid** [comment]
@josephsweeney Yup, got them now thanks!

**Pull request merged**